tests/core/test_graph.py

```diff
@@ -128,6 +128,39 @@ def test_legacy_constructor_warning():
         StellarGraph(nx.Graph())
 
 
+def test_graph_constructor_extra_nodes_in_edges():
+    nodes = pd.DataFrame(np.ones((5, 1)), index=[0, 1, 2, 3, 4])
+    edges = {
+        "a": pd.DataFrame({"source": [1], "target": [0]}, index=[0]),
+        "b": pd.DataFrame({"source": [4, 5], "target": [0, 2]}, index=[1, 2]),
+    }
+
+    with pytest.raises(ValueError):
+        g = StellarGraph(nodes=nodes, edges=edges)
+
+    # adding an extra node should fix things
+    nodes = pd.DataFrame(np.ones((6, 1)), index=[0, 1, 2, 3, 4, 5])
+    g = StellarGraph(nodes=nodes, edges=edges)
+
+    # removing the bad edge should also fix
+    nodes = pd.DataFrame(np.ones((5, 1)), index=[0, 1, 2, 3, 4])
+    edges = {
+        "a": pd.DataFrame({"source": [1], "target": [0]}, index=[0]),
+        "b": pd.DataFrame({"source": [4], "target": [0]}, index=[1]),
+    }
+
+
+def test_graph_constructor_nodes_from_edges():
+    edges = {
+        "a": pd.DataFrame({"source": [1], "target": [0]}, index=[0]),
+        "b": pd.DataFrame({"source": [4, 5], "target": [0, 2]}, index=[1, 2]),
+    }
+
+    g = StellarGraph(nodes=None, edges=edges)
+
+    assert sorted(g.nodes()) == [0, 1, 2, 4, 5]
+
```

Comment:
Also, just in case the code changes dramatically, it might be good to check the node-type defaulting correctly here:

```python
g = StellarGraph(edges=edges, node_type_default="abc")
assert g.node_types == {"abc"}
```